tests/pauli_rotations/circuit_test.py

```diff
@@ -57,6 +65,91 @@ def test_count_rotations_by(circuit, fraction, expected):
     assert circuit.count_rotations_by(fraction) == expected
 
 
-# def test_remove_y_operators_from_circuit(output, expected):
-#     raise NotImplementedError
-#     # assert output == expected
+@pytest.mark.parametrize(
+    "rotation1, rotation2",
+    [
+        (([X], Fraction(1, 4)), ([I], Fraction(1, 4))),
+        (([I, X], Fraction(1, 4)), ([Y, X], Fraction(-1, 8))),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Y], Fraction(1, 8))),
+        (([I, Z, I, X], Fraction(-1, 4)), ([Z, X, I, Z], Fraction(-1, 8))),
+    ],
+)
+def test_commute_rotation_to_rotation_commuting(rotation1, rotation2):
+    test_circuit = PauliOpCircuit(len(rotation1[0]))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation1))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation2))
+    test_circuit.commute_pi_over_four_rotation(0)
+
+    assert test_circuit.ops[0].ops_list == rotation2[0]
+    assert test_circuit.ops[0].rotation_amount == rotation2[1]
+    assert test_circuit.ops[1].ops_list == rotation1[0]
+    assert test_circuit.ops[1].rotation_amount == rotation1[1]
+
+
+@pytest.mark.parametrize(
+    "rotation1, rotation2, resultant_rotation",
```

Comment:
I might suggest renaming `resultant_rotation` to something that makes it clear that it is what rotation2 becomes, so maybe `first_resulting_rotation` ? because now it looks like two rotations are becoming one, but that is not the case.

Then also possibly add a comment documenting the fact that the second resulting rotation stays unchanged. I see you are checking it in the test case but it's not immediately obvious
``````

Reply:
Good point - I will add that in.

Reply:
comments added